### DISCUSSION TITLE
set padding only for the first page

### DISCUSSION THREAD

**Discussion by ballotade** [dompdf/dompdf#3032] [Q&A]
Hello,
I placed a footer with position: absolute; bottom: 0; on the first page. 

To avoid that the main Text overwrites the footer, i tried to set the margin-bottom property at `@page`. But this also lifts the footer. How can I avoid that the main text overwrites the footer on the first page? I tried  `@page { padding-bottom: 50mm; }` but this does not change anything.
is there a possibility to change properties only on the first `@page`? like `@page:first {...}`?

I hope that you can understand what I mean.
Greetings from Germany

**Comment by bsweeney** [comment]
The bottom position is based on the body boundary. You can ensure that positioned content does not overlap normally flowed content by positioning your content outside the bounds of the body. So if your header div is 50px tall, you can ensure it doesn't overlap any content by setting the bottom to -50px.

You can set a different margin on the first page, you just need a space in your selector: `@page :first`.

```
<!DOCTYPE html>
<html>
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8"/>
  <style>
    @page {
      margin: 50px;
    }
    @page :first {
      margin: 100px;
    }
    body {
      background-color: gray;
    }
    .foot {
      position: fixed;
      inset: auto 0px -50px 0px;
      background-color: #ff33ff33;
      height: 50px;
    }
  </style>
</head>
<body>
  <div class="foot"></div>
  <div style="page-break-before: always;"></div>
</body>
</html>
```

**Reply by ballotade** [reply]
Thank you. It worked after I used only one mesurement.
In the first time I used mm and pt in one margin statement and there it was not crrect. What ist the conversion factor fromm mm to pt?

**Reply by bsweeney** [reply]
mm has a fixed translation of [{length} * 72 / 25.4](https://github.com/dompdf/dompdf/blob/v2.0.1/src/Css/Style.php#L1061-L1063)
